### ISSUE TITLE
YF doesn't work for the cv task

### ISSUE THREAD

**Issue by EdwardTyantov** [JianGoForIt/YellowFin_Pytorch#5]
Hi, I tried your optimizer instead of SGD for this challenge https://www.kaggle.com/c/planet-understanding-the-amazon-from-space and get this kind of train/valid curves https://s.mail.ru/BFbQ/1K6w1bqD7 , which is obv awful (tried different setups but no success). SGD & plateau scheduler reach 0.08258 val loss. 

What data do you need to investigate such bad performance ? For example learning rates changes between 0.1 and 3 (!) .

**Comment by JianGoForIt** [comment]
Hi @EdwardTyantov, 

Thanks for trying out our optimizer. 

Seen from the figure you provided, I think there are two things you want to try out.

1. There are exploding or zero gradient in the middle of training, which drive the learning rate and momentum changing crazily. You might want to do gradient clipping to avoid this, Please refer to the solutions in the discussion here https://github.com/JianGoForIt/YellowFin_Pytorch/issues/1

2. It seems there is an initial increasing of training loss, it might help if you play with the initial learning rate a bit to eliminate initial increasing of the training loss.

Hope it helps and please let us know whether it helps or not.

**Comment by EdwardTyantov** [comment]
Thank you for the quick reply!
1. OK. I'll try clipping, will report on the results here )
2. I've already played with it - wide range on which SGD showed acceptable performance.

**Comment by jtoy** [comment]
@EdwardTyantov please share your findings here!

**Comment by JianGoForIt** [comment]
Closed due to lack of activity. Feel free to reopen if necessary.
